### ISSUE TITLE
Possible to backprop through UMAP encoding/sensitivity analysis?

### ISSUE THREAD

**Issue by NMRobert** [lmcinnes/umap#555]
Hi,

If I constructed a pipeline consisting first of a UMAP embedding R^N->R^M followed by some differentiable mapping R^M->R1 (say, in the case of a simple binary classifier), is there a way to backprop 'through' the encoder such that one could perform something like integrated gradients for sensitivity analysis/feature attribution?

I'm very curious as to what people are doing for explainability when using UMAP. Of course it is always possible to do some sort of perturbation sensitivity analysis or kernelSHAP (etc), but this tends to not be super computationally efficient.

Thank you!

**Comment by jc-healy** [comment]
Hi there,

I'm always excited to see people trying to come up with better ways to
handle expandability (especially in unsupervised cases).  Have you had a
look at the new parametric UMAP?  It uses a neural network to learn a
mapping from the original space to a low dimensional one using the UMAP
objective function.  It sounds like it's what you would need to get started
with this project.
Check out the read the docs page here:
https://umap-learn.readthedocs.io/en/latest/parametric_umap.html
And the paper over on arxiv: https://arxiv.org/abs/2009.12981

**Comment by NMRobert** [comment]
@jc-healy Thank you, that looks really promising. I'll have a go at it and see if I can create some illustrative examples :) I'm not sure if it would be more appropriate to close this github issue, but please feel free to do so if it will help keep things tidy.
